src/utils/EmailType.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { GenericIdentifierType } from './GenericIdentifierType';
+import { FunctionalComponent, h } from '@stencil/core';
+
+/**
+ * This class specifies a custom renderer for Email addresses.
+ * @extends GenericIdentifierType
+ */
+export class EmailType extends GenericIdentifierType {
+  getSettingsKey(): string {
+    return 'EmailType';
+  }
+
+  hasCorrectFormat(): boolean {
+    const regex = /^((\(.*\))?(\w|-|\+|_|\.)+(\(.*\))?@((((\(.*\))?\w+(\(.*\))?\.)+(\(.*\))?\w+(\(.*\))?\w+)|(\[((IPv6:((\w+:+)+\w+))|(\d+\.)+\d+)\]))\s*(,|$)\s*)+$/
+    return regex.test(this.value);
+  }
+
+  init(): Promise<void> {
+    return;
+  }
+
+  isResolvable(): boolean {
+    return false;
+  }
+
+  renderPreview(): FunctionalComponent<any> {
+    return (
+      <span>
+        {React.Children.map(this.value.split(new RegExp(/\s*,\s*/)), (email, i) => {
```

Comment:
What do you need React for? You can simply use “normal” JS and the tools Stencil provides you. This is shown in many places of the Documentation eg. here[`{this.mailaddresses.map((email) => (
          <a href="...">...</a>
        )}`](https://stenciljs.com/docs/templating-jsx#dealing-with-children).

Reply:
done 9f953ec47359aa05a3d2e7ecee5863599dc1514a